compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -303,7 +306,7 @@ object Implicits:
    */
   class ContextualImplicits(
       val refs: List[ImplicitRef],
-      val outerImplicits: ContextualImplicits,
+      val outerImplicits: ContextualImplicits | Null,
```

Comment:
Is this ever null other than for NoContext? If not, is there anything cleaner we could do?